src/psyclone/psyad/tl2ad.py

```diff
@@ -128,9 +129,14 @@ def generate_adjoint_str(tl_fortran_str, active_variables,
         Config.get().api = api
         ad_psyir = generate_lfric_adjoint(tl_psyir, active_variables)
         if create_test:
+            test_name = "adjoint_test"
+            # Test filenames for LFRic API must be of form foo_alg_mod.[Xx]90
+            if test_filename:
+                test_name = test_filename.split("_mod.")[0]
```

Comment:
Previously, we didn't care what filename the user had supplied. Now we require it to be of a particular form (if the LFRic API is specified) and therefore we need to validate it and raise a suitable error if it doesn't conform. It would probably be best to do this back where the other validation of the command-line arguments is performed (and then add a comment here to say that this assumption has been validated previously).

Reply:
The new filename validation will require a new test too - please add this in whichever file we test the psyad command-line arguments.

Reply:
The refactoring into `main.py` and addition of a new test was addressed in 73d818f2ad29ad1c7bbf606048890bb5ba123ce2. I also added a new test in 81234adc912342788a83581acdce0b07fc04b347 which uses a non-conformant name and checks to see if the relevant error is present.